lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -264,6 +266,12 @@ void IMDeadCodeElimPass::markBlockExecutable(Block *block) {
     else if (hasUnknownSideEffect(&op))
       markUnknownSideEffectOp(&op);
 
+    // Recursively mark any blocks contained within these operations as
+    // executable.
+    for (auto &region : op.getRegions())
+      for (auto &block : region.getBlocks())
+        markBlockExecutable(&block);
```

Comment:
This recursion can blow up the stack for extremely deeply-nested blocks. I don't think that's an actual issue in practice, especially since we probably parse these blocks with recursive function calls in the parser, so this doesn't add new scaling issues. WDYT?

Reply:
I was worried about this when reading it... It is narrowly, recursion over regions (which is allowed by the MLIR developer policy). However, I agree that it is risky to do it this way. I can see about rewriting this to be iterative (here or in a follow-on).

Reply:
:+1: Using recursion on blocks feels fine :+1: